modules/asset-manager-impl/src/main/java/org/opencastproject/assetmanager/impl/AssetManagerImpl.java

```diff
@@ -947,49 +949,54 @@ public void repopulate() throws IndexRebuildException {
       securityService.setOrganization(defaultOrg);
       securityService.setUser(defaultSystemUser);
 
+      int offset = 0;
+      int total = (int) countEvents(null);
       final AQueryBuilder q = createQuery();
-      final RichAResult r = enrich(q.select(q.snapshot()).where(q.version().isLatest()).run());
-      final int total = r.countSnapshots();
-      logIndexRebuildBegin(logger, index.getIndexName(), total, "snapshot(s)");
+      RichAResult r;
       int current = 0;
-      int n = 16;
-      var updatedEventRange = new ArrayList<Event>();
+      do {
+        r = enrich(q.select(q.snapshot()).where(q.version().isLatest()).page(offset, PAGE_SIZE).run());
+        offset += PAGE_SIZE;
+        logIndexRebuildBegin(logger, index.getIndexName(), total, "snapshot(s)");
```

Comment:
This log line should be outside the loop. Otherwise, you will see this during every iteration:
```
Starting update of the Elasticsearch index from service 'AssetManager' with ... snapshot(s)
```